crates/solver/src/driver.rs

```diff
@@ -409,6 +410,14 @@ impl Driver {
     }
 
     pub async fn single_run(&mut self) -> Result<()> {
+        let id = self.next_auction_id();
+        // extra function so that we can add span information
+        self.single_run_(id)
+            .instrument(tracing::debug_span!("single_run", id))
```

Comment:
uber-nano-pico-nit:
```suggestion
            .instrument(tracing::debug_span!("auction", id))
```

Wouldn't this look nicer in the logs as it would print `auction=42`? Either way, I think its fine as is.